### PR TITLE
refactor: replace read-pkg-up with escalade

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://packagephobia.com/result?p=read-pkg-up

read-pkg-up is quite big package with many dependencies.
It solves cases not necessary for this project like pretty json errors
and normalisation of package.json data.

In this diff I replaced it with dependency-free escalade and parsed json
with native JSON.parse.

This will let users to have faster installation and less bloated
node_modules.

This PR contains:

- [ ] bugfix
- [ ] feature
- [x] refactor
- [ ] tests
- [ ] documentation
- [ ] metadata

### Breaking Changes?

- [ ] yes
- [x] no

### CODE REVIEW

**Review by PlayMa256 — APPROVED** [review]
Less deps is always a good thing

File: lib/plugins/ramdisk.js [hunk 1]
```diff
@@ -14,12 +14,30 @@ const { symlinkSync } = require('fs');
...
 
 const isCwd = require('is-path-cwd');
-const readPkgUp = require('read-pkg-up');
+const escalade = require('escalade/sync');
 const rm = require('rimraf');
 const { WebpackPluginRamdisk } = require('webpack-plugin-ramdisk');
 
 const { PluginExistsError, RamdiskPathError } = require('../errors');
 
+function readPkgName() {
```
Comment (shellscape) [inline lib/plugins/ramdisk.js:23]:
```suggestion
const readPkgName = () => {
```

We should export this so it can be tested

**Review by shellscape — COMMENTED** [review]
I'm good overall with the change, but we need to get test coverage around this now. The one downside to removing dependencies that take care of this type of thing for you is that we now have to own more code, and that means tests.

**Comment by TrySound** [comment]
Heh, I just found currently pkg.name is always null so we always get ramdom directory name. pkg.name is incorrect. Should've been pkg.packageJson.name.

**Comment by TrySound** [comment]
Current tests covers logic around escalade. Though the code with generating id cannot be reached. Not sure how to cover it with tests as well.

File: lib/plugins/ramdisk.js [hunk 2]
```diff
@@ -10,16 +10,34 @@
...
+    return null;
+  }
+  try {
+    const pkg = JSON.parse(readFileSync(pkgPath, 'utf-8'));
+    return pkg.name;
+  } catch (error) {
+    return null;
+  }
+};
+
```
Comment (shellscape) [inline lib/plugins/ramdisk.js:40]: we're going to need test coverage of this method before we can merge

Comment (TrySound) [reply lib/plugins/ramdisk.js:40]: As I mentioned above. This is covered by existing tests. See https://github.com/shellscape/webpack-plugin-serve/pull/187/files#diff-817a4e640cdcf9964df2158c8020f627R33-R54

Though I can add test with  broken package.json. This way random name can be tested too.

Comment (TrySound) [reply lib/plugins/ramdisk.js:40]: No, broken package.json will not work because webpack-nano failed as well for some reason.

**Comment by TrySound** [comment]
I don't understand what is the problem
image

**Comment by bebraw** [comment]
Looks good. Should we merge (squash please)?

**Comment by TrySound** [comment]
@bebraw Done

**Comment by bebraw** [comment]
Ok, merged.

We can even squash on merge without any extra work (one of the options) so maybe we'll use that in the future. 👍

**Comment by TrySound** [comment]
I enforced only squash commits at my work.

**Comment by bebraw** [comment]
> I enforced only squash commits at my work. Can be configured in settings.

That's a good idea.

@shellscape Do you want to set that up?

**Pull request merged**